app/main.py

```diff
@@ -2,5 +2,13 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cashe_dict = {}
+
+    def inner(*args, **kwargs) -> Callable:
+        if args in cashe_dict:
+            print("Getting from cache")
+        else:
+            print("Calculating new result")
+            cashe_dict[args] = func(*args)
```

Comment:
The function 'func' is called with '*args' only. If 'func' can also accept keyword arguments, you should pass '**kwargs' as well.